server/personality/personality.service.ts

```diff
@@ -79,33 +79,51 @@ export class PersonalityService {
      * @param personality PersonalityBody received of the client.
      * @returns Return a new personality.
      */
-    create(personality) {
+    async create(personality) {
         try {
-            personality.slug = slugify(personality.name, {
-                lower: true,     // convert to lower case, defaults to `false`
-                strict: true     // strip special characters except replacement, defaults to `false`
-            })
-            const newPersonality = new this.PersonalityModel(personality);
-            this.logger.log(
-                `Attempting to create new personality with data ${personality}`
-            );
-
-            const user = this.req.user
-
-            const history = this.history.getHistoryParams(
-                newPersonality._id,
-                TargetModel.Personality,
-                user,
-                HistoryType.Create,
-                personality
-            )
-
-            this.history.createHistory(history)
-
-            return newPersonality.save();
+            const personalityExisted = await this.getDeletedPersonalityByWikidata(personality.wikidata)
+            if(personalityExisted) {
+                return personalityExisted
+            } else {
+                personality.slug = slugify(personality.name, {
+                    lower: true,     // convert to lower case, defaults to `false`
+                    strict: true     // strip special characters except replacement, defaults to `false`
+                })
+                const newPersonality = new this.PersonalityModel(personality);
+                this.logger.log(
+                    `Attempting to create new personality with data ${personality}`
+                );
+    
+                const user = this.req.user
+    
+                const history = this.history.getHistoryParams(
+                    newPersonality._id,
+                    TargetModel.Personality,
+                    user,
+                    HistoryType.Create,
+                    personality
+                )
+    
+                this.history.createHistory(history)
+    
+                return newPersonality.save();
+            }
         } catch (err) {}
     }
 
+    async getDeletedPersonalityByWikidata(wikidata) {
+        const deletedPersonalities = await this.PersonalityModel.findDeleted(true)
```

Comment:
What do you think about executing a find and pass the wikidata as a paramter? Like that:
```
this.personalityModel.find({ wikidata, deleted: true})
```

That way you don't need to filter the array afterwards. The wrong choice here is to get all deleted and iterate over them, what if you have 1 million deleted personalities? This would cause a performance penalty and this endpoint could break or become very slow. Furthermore, this could also become a DDOS breach.